xonsh/tools.py

```diff
@@ -794,11 +794,15 @@ def ensure_int_or_slice(x):
     elif is_int(x):
         return x
     # must have a string from here on
-    if ':' in x:
-        x = x.strip('[]()')
-        return slice(*(int(x) if len(x) > 0 else None for x in x.split(':')))
-    else:
-        return int(x)
+    try:
+        if ':' in x:
+            x = x.strip('[]()')
+            return slice(*(int(x) if len(x) > 0 else None for x in x.split(':')))
+        else:
+            return int(x)
+    except ValueError:
+        print("Could not convert index or slice to int.")
+        return False
```

Comment:
This should simply fail if incorrect input is passed to it.  It is the job of an `is_int_or_slice()` function to makes sure that x is valid.